app/src/main/java/org/mozilla/fenix/home/recenttabs/controller/RecentTabController.kt

```diff
@@ -35,15 +40,27 @@ interface RecentTabController {
  */
 class DefaultRecentTabsController(
     private val selectTabUseCase: SelectTabUseCase,
-    private val navController: NavController
+    private val navController: NavController,
+    private val metrics: MetricController,
+    private val store: BrowserStore
 ) : RecentTabController {
 
     override fun handleRecentTabClicked(tabId: String) {
+
+        when (tabId) {
```

Comment:
nit: We could save some tabs iterations if we only check for `inProgressMediaTab` and record `OpenRecentTab` in all other cases.